backends/advanced/src/advanced_omi_backend/services/transcription/__init__.py

```
@@ -1,115 +1,276 @@
+        self._streams.pop(client_id, None)
+
+        if not isinstance(final, dict):
+            return {"text": "", "words": [], "segments": []}
+        return {
+            "text": _dotted_get(final, extract.get("text")) if extract else final.get("text", ""),
+            "words": _dotted_get(final, extract.get("words")) if extract else final.get("words", []),
+            "segments": _dotted_get(final, extract.get("segments")) if extract else final.get("segments", []),
+        }
+
+
+def get_transcription_provider(provider_name: Optional[str] = None, mode: Optional[str] = None) -> Optional[BaseTranscriptionProvider]:
```

Ruff (0.14.8)

Unused function argument: `provider_name`

(ARG001)